server.go

```diff
@@ -1859,11 +1876,13 @@ func (s *Server) serveConn(c net.Conn) error {
 	ctx := s.acquireCtx(c)
 	ctx.connTime = connTime
 	isTLS := ctx.IsTLS()
+	if s.Trace.AcquiredContext != nil {
+		s.Trace.AcquiredContext(ctx)
```

Comment:
looking at all of these hooks, I've got some questions:

- what will happen with a server, if s.Trace.AcquiredContext will block for 10 minutes under load?
- how much this will cost (in performance terms) to a person, who will not use this tracing?
- are we sure that we're not introducing unnecessary hooks?

Reply:
>what will happen with a server, if s.Trace.AcquiredContext will block for 10 minutes under load?

If a user of the tracing functionality writes blocking code, it will absolutely affect the overall performance of the server. I can highlight in the documentation that all hooks are processed synchronously in a blocking manner.

>how much this will cost (in performance terms) to a person, who will not use this tracing?

For use cases that do not utilize any kind of tracing this introduces a few `s.Trace.<hook> != nil` checks. I can try to benchmark this but a simple comparison on a pointer should be negligible in terms of performance.

>are we sure that we're not introducing unnecessary hooks?

 I only included hooks for things that I started using in my production use case and the ones required to deprecate `ConnState`. If anything, there is probably room for additional hooks but I will leave that up to other contributors as it is hard to foresee other use cases.